7-base_geometry.py module
============================================================


Using ``integer_validator() `` from ``BaseGeometry()``

----------
TEST CASES
----------


VALID VALUE TO INTEGER VALIDATOR
=========================

::
	>>> BaseGeometry = __import__('7-base_geometry').BaseGeometry
	>>> bg = BaseGeometry()
	>>> bg.integer_validator("my_int", 12)
	12
	>>> bg.integer_validator("width", 89)
	89

INVALID VALUES TO INTEGER VALIDATOR
===================================
::
	>>> bg.integer_validator("age", 0)
	Traceback (most recent call last):
	ValueError: age must be greater than 0
	>>> bg.integer_validator("distance", -4)
	Traceback (most recent call last):
	ValueError: distance must be greater than 0
	>>> bg.integer_validator("name", "John")
	Traceback (most recent call last):
	TypeError: name must be an integer
	>>> bg.integer_validator("vector", [2,35])
	Traceback (most recent call last):
	TypeError: vector must be an integer
